### PR TITLE
Add other required clients

### PR DESCRIPTION
**Pull request by fiunchinho**
Towards https://github.com/giantswarm/giantswarm/issues/7982

These clients will be used to spin up a bastion instance during Azure e2e tests.

### CODE REVIEW

**Review by xh3b4sd — APPROVED** [review]

File: azure/azure.go [hunk 1]
```diff
@@ -25,8 +27,16 @@ var (
...
 type Client struct {
+	InterfacesClient                *network.InterfacesClient
+	IPAddressesClient               *network.PublicIPAddressesClient
+	ResourceGroupsClient            *resources.GroupsClient
+	SecurityGroupsClient            *network.SecurityGroupsClient
+	SecurityRulesClient             *network.SecurityRulesClient
+	SubnetsClient                   *network.SubnetsClient
 	VirtualMachineScaleSetsClient   *compute.VirtualMachineScaleSetsClient
 	VirtualMachineScaleSetVMsClient *compute.VirtualMachineScaleSetVMsClient
+	VirtualNetworksClient           *network.VirtualNetworksClient
```
Comment (MarcelMue) [inline azure/azure.go]: Order is wrong - `VirtualM..` should be before `VirtualN...`

**Review by MarcelMue — APPROVED** [review]
Smallest nit of my life. Are these all clients?

**Comment by fiunchinho** [comment]
> Smallest nit of my life. Are these all clients?

Yes, they are. Why?

**Comment by MarcelMue** [comment]
> > Smallest nit of my life. Are these all clients?
> 
> Yes, they are. Why?

I thought there were even more that might be used somewhere. All good.

**Comment by fiunchinho** [comment]
There might be. I'm using these new clients in new code.

**Pull request merged**